addons/l10n_ec_website_sale/models/sale_order.py

```diff
@@ -0,0 +1,25 @@
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+from odoo import models
+
+class SaleOrder(models.Model):
+    _inherit = 'sale.order'
+
+    def _create_invoices(self, grouped=False, final=False, date=None):
+        """ Create invoice(s) for the given Sales Order(s).
+
+        :param bool grouped: if True, invoices are grouped by SO id.
+            If False, invoices are grouped by keys returned by :meth:`_get_invoice_grouping_keys`
+        :param bool final: if True, refunds will be generated if necessary
+        :param date: unused parameter
+        :returns: created invoices
+        :rtype: `account.move` recordset
+        :raises: UserError if one of the orders has no invoiceable lines.
+        """
+        moves = super()._create_invoices(grouped, final, date)
+        for move in moves:
+            if move.transaction_ids:
+                sri_payment_methods = move.transaction_ids.mapped('payment_method_id.l10n_ec_sri_payment_id')
+                if len(sri_payment_methods) == 1:
+                    move.l10n_ec_sri_payment_id = sri_payment_methods
```

Comment:
The post() on the invoice validates that the invoice has a l10n_ec_sri_payment_id set raising an error if not. If the payment method is not set then the whole transaction will roll back.

Maybe we should warn the administrator about the missing configuration of the field payment_method_id.l10n_ec_sri_payment_id? or maybe we should not show the payment method at all.

The big problem with this is to have a transaccion is rolled back AFTER a succesfull charge to the credit card, it will cause serious complains.

Another options are:
- To don't require the l10n_ec_sri_payment_id in the post() of the account.move... at least for ecommerce
- To make the field l10n_ec_sri_payment_id required in the payment.method when country is EC

Reply:
If I understand you correctly you think that hiding payment methods that do not have the `l10n_ec_sri_payment_id` field set would be good enough? (see above)

Reply:
@svfu-odoo Maybe doing the field required? So when the user creates the eCommerce Payment Methos, is required to allocate this value. Or we can map all of other Payment Methods (besides the 2 described above: Card and Wire Transfer) using the SRI Payment method "No use of the financial system"

What do you think @pepetreshere ?
cc' @jco-odoo

Reply:
Yes, @svfu-odoo your solution of hidding the payment method is enough, cool!, so we could close this comment